cocos/base/CCUserDefault-android.cpp

```diff
@@ -519,6 +519,17 @@ void UserDefault::flush()
 {
 }
 
+bool UserDefault::deleteValueForKey(const char* key)
+{
+    // check the params
+    if (!key)
```

Comment:
It's better to output some debug info to tell why it fails.
Like `CCLOG("key can't be nullptr!")`.